scripts/set-version.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+set -exo pipefail
```

Comment:
I think this line deals with pipeline failure but what does this line do?

Reply:
It's a one liner that does a couple things. See: https://gist.github.com/mohanpedala/1e2ff5661761d3abd0385e8223e16425